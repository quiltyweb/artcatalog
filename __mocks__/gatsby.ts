import React from "react";

const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(({ to, ...rest }) =>
    React.createElement("a", {
      ...rest,
      href: to,
    })
  ),
  Slice: jest.fn().mockImplementation(({ alias, ...rest }) =>
    React.createElement("div", {
      ...rest,
      "data-test-slice-alias": alias,
    })
  ),
  StaticQuery: jest.fn(),
};
