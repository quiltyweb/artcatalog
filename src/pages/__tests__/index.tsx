import React from 'react';
import * as Gatsby from 'gatsby';
import renderer from 'react-test-renderer';

import IndexPage from '../index';

beforeEach(() => {
  jest.clearAllMocks();
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('IndexPage', () => {
  it('renders correctly', () => {
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: 'My Title',
        },
      },
    }));
    const tree = renderer.create(<IndexPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
