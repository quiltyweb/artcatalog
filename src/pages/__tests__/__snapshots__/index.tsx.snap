// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
<main
  className="sc-bdnxRM eevvKH"
>
  <title>
    Welcome
     | 
    My Title
     
  </title>
  <p>
    My Title
  </p>
  <nav>
    <ul
      className="nav-links"
    >
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link-text"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="nav-link-item"
      >
        <a
          className="nav-link-text"
          href="/about"
        >
          About
        </a>
      </li>
    </ul>
  </nav>
  <h1
    className="heading"
  >
    Welcome
  </h1>
  <p>
    ~ Work in progress ~
  </p>
</main>
`;
